File: Searching/largestThree.cpp
#include<bits/stdc++.h>
using namespace std;

void largestThree(int arr[],int n){
    int max1=INT_MIN,max2=INT_MIN,max3=INT_MIN;
    if(n<3)
        cout<<"Invalid input"<<endl;
     for(int i=0;i<n;i++){
        if(arr[i]>max1){
            max3=max2;
            max2=max1;
            max1=arr[i];
        }
        else if(arr[i]>max2 && arr[i]!=max1){
            max3=max2;
            max2=arr[i];
        }
        else if(arr[i]>max3 && arr[i]!=max2){
            max3=arr[i];
        }
    }
    cout<<"max1: "<<max1<<" max2: "<<max2<<" max3: "<<max3<<endl;
}

int main(){
    int arr[]={10, 4, 3, 50, 23, 90};
    int n=sizeof(arr)/sizeof(arr[0]);
    largestThree(arr,n);
    return 0;
}

File: Searching/largestThree.java
import java.util.*;
public class largestThree {
    static void largestThree(int arr[],int n){
        int first=Integer.MIN_VALUE,second=Integer.MIN_VALUE,third=Integer.MIN_VALUE;
        if(n<3){
            System.out.println("Invalid input");
            return;
        }
        for(int i=0;i<n;i++){
            if(arr[i]>first){
                third=second;
                second=first;
                first=arr[i];
            }
            else if(arr[i]>second && arr[i]!=first){
                third=second;
                second=arr[i];
            }
            else if(arr[i]>third && arr[i]!=second){
                third=arr[i];
            }
        }
        System.out.println("first: "+first+" second: "+second+" third: "+third);
    }

    public static void main(String[] args) {
        int arr[]={10,2,3,5,50,52};
        int n=arr.length;
        largestThree(arr,n);

    }
}
